lib/firewall/firewall.cc

```diff
@@ -748,3 +753,45 @@ void firewall_remove_udpipv6_remote_endpoint(uint8_t *remoteAddress,
 		  IPProtocolNumber::UDP, *copy, localPort, remotePort);
 	}
 }
+
+bool ethernet_driver_start(std::atomic<uint32_t> *state)
+{
+	if (TCPIPRestartState == nullptr)
+	{
+		if (!CHERI::check_pointer<CHERI::PermissionSet{
+		      CHERI::Permission::Load, CHERI::Permission::Global}>(
+		      state, sizeof(*state)))
+		{
+			Debug::log("Invalid TCP/IP state pointer {}", state);
+			return false;
+		}
+		TCPIPRestartState = state;
+	}
```

Comment:
You mean checking that `TCPIPRestartState == state`? We could, but I am not sure how to react to a mismatch. The value of this pointer should never change because it is a global. Thus any mismatch should either be a bug or a malicious attempt at corrupting the firewall. Do you think it is worth adding the branch just for a debug print?

Reply:
I wasn't sure how we should react on an `else` branch, either.  But if the `TCPIPRestartState` object is just something that needs to be shared between the firewall and the network compartment, could we statically arrange for that sharing rather than pass it across the compartment boundary?

That is, could we have one or the other export it and the other import it and rely on the `cheriot-audit` tool to make sure nobody else is trying to get their hands on it?  If so, I think we could do away with the parameter, and so this test, entirely?

Reply:
We don’t (yet) have a mechanism to export globals from one compartment to another.

Reply:
In lieu of exporting globals, could we export a getter for the moment?

Reply:
Sure! We could also export a getter. I suspect that it would take more code size though?